### DISCUSSION TITLE
Is a FIN-related implementation mandatory?

### DISCUSSION THREAD

**Discussion by hyn2028** [ANLAB-KAIST/KENSv3#75] [Project Q&A 2. Transport Layer (TCP)] [answered]
TCP's connection termination procedure seems much more complex to implement than a 3-way handshake. 
Is implementation of this mandatory in PA2? If all the test cases pass, can we not implement it?

Thank you.

**Comment by Pusnow** [comment, marked as answer]
Passing all test cases is enough for project assignment, but we strongly recommend to implement it.
